File: frontend/board/src/app/modules/core/models/account.ts
export enum AccountRole {
  ADMIN = 'admin',
  EMPLOYEE = 'employee',
  VIEWER = 'viewer',
}

export interface Account {
  id: number;
  name: string;
  username: string;
  isActive: boolean;
  avatarUrl: string | null
  authProvider: string
  role: AccountRole
}

export interface AccountAuthResult {
  result?: boolean;
  message?: string;
  user: Account | null;
  errors?: string[];
}

export interface OnlineAccount {
  id: number;
  name: string;
  avatarUrl: string | null;
}

export interface RegistrationRequest {
  username: string;
  password: string;
  first_name?: string;
  last_name?: string;
}

export interface RegistrationResult {
  user: Account;
  message: string;
}

File: backend/internal/domain/models/account.go
package models

type AccountId uint

type AuthProvider string

const (
	AuthProviderPassword AuthProvider = "password"
	AuthProviderGitLab   AuthProvider = "gitlab"
)

type AccountRole string

const (
	AccountRoleAdmin    AccountRole = "admin"
	AccountRoleEmployee AccountRole = "employee"
	AccountRoleViewer   AccountRole = "viewer"
)

type Account struct {
	IsActive     bool
	Id           AccountId
	Username     string
	Password     string
	Name         string
	GitlabID     *UserId
	AuthProvider AuthProvider
	AvatarURL    string
	Role         AccountRole
	JwtSalt      string
}
